Sources/Request+Locale.swift

```diff
@@ -0,0 +1,45 @@
+import Vapor
+import Lingo
+
+extension Request {
+    var locale: String {
+        if let session = try? session(), let sessionlocale = session["locale"] {
+            return sessionlocale
+        } else {
+            // Parse locale, ordered from general case to special case
+            // If nothing helps.. use "en"
+            var locale: LocaleIdentifier = "en"
+            
+            // If Lingo has defaultLocale, use that
+            if let lingo = try? make(Lingo.self) {
+                locale = lingo.defaultLocale
+            
+                // If headers provide information, use those
+                if var langHeader = http.headers.firstValue(name: .acceptLanguage) {
+                    langHeader = langHeader.components(separatedBy: .whitespaces).joined()
+                    let langValues = langHeader.components(separatedBy: ",").map { HeaderValue.parse($0) ?? HeaderValue($0, parameters: ["q":"0"]) }
+                    let availableLocales = (try? lingo.dataSource.availableLocales()) ?? [locale]
+                    var localeWeight = 0.0
+                    for lang in langValues {
+                        let q = Double(lang.parameters["q"] ?? "1") ?? 1
+                        if localeWeight < q {
+                            if availableLocales.contains(lang.value) {
+                                localeWeight = q
+                                locale = lang.value
+                            } else if let baseLang = lang.value.split(separator: "-").first, availableLocales.contains(String(baseLang)) {
+                                localeWeight = q-0.01
+                                locale = String(baseLang)
+                            }
+                        }
+                    }
+                }
+            }
+            
+            // Save locale into session and return
+            if let session = try? session() {
+                session["locale"] = locale
```

Comment:
This is something that I am not really happy about. In my opinion, it should not be responsibility of a `Request` to modify session. Also from a more general perspective, accessing `request.session` should not have a side effect of modifying a session state.

I believe this kind of logic should be on much higher level, probably some controller could decide to modify a session. For that reason you might which to expose a method that would allow it to get the locale of a request.

Reply:
I get your point and I agree with you. The side effect is surprising at this point. I removed that part und mentioned it in the readme. I think, the locale changes only on purpose, so it can be put into the controller, that handles the change of the locale.
Now I only think about performance. 🤔 When this property is called for every localization, it might be a bit too much. What do you think?